bindings/rust/bench/benches/resumption.rs

```diff
@@ -0,0 +1,84 @@
+use bench::{
+    CipherSuite, CryptoConfig, HandshakeType, KXGroup, S2NConnection, SigType, TlsConnPair,
+    TlsConnection,
+};
+use criterion::{
+    criterion_group, criterion_main, measurement::WallTime, BatchSize, BenchmarkGroup, Criterion,
+};
+
+fn bench_handshake_pair<T: TlsConnection>(
+    bench_group: &mut BenchmarkGroup<WallTime>,
+    sig_type: SigType,
+) {
+    // generate all harnesses (TlsConnPair structs) beforehand so that benchmarks
+    // only include negotiation and not config/connection initialization
+    for handshake in [HandshakeType::Resumption, HandshakeType::ServerAuth] {
+        bench_group.bench_function(format!("{:?}-{}", handshake, T::name()), |b| {
+            b.iter_batched_ref(
+                || {
+                    TlsConnPair::<T, T>::new(
+                        CryptoConfig::new(CipherSuite::default(), KXGroup::default(), sig_type),
+                        handshake,
+                        Default::default(),
+                    )
+                },
+                |conn_pair_res| {
+                    if let Ok(conn_pair) = conn_pair_res {
+                        let _ = conn_pair.handshake();
+                    }
+                },
+                BatchSize::SmallInput,
+            )
+        });
+    }
+}
+
+fn bench_handshake_server_1rtt<T: TlsConnection>(
```

Comment:
What is this bench for? It seems like bench_handshake_pair is already measuring the cost of a full handshake versus the cost of a resumed handshake.

Reply:
Certain customer might be more interested in the savings of a particular connection. For example, a customer who runs a large fleet of servers could use the benchmark to see what kinds of savings they might realize on their fleet.